rpc-java-client/src/main/java/org/sonarsource/sonarlint/core/rpc/client/SloopLauncher.java

```diff
@@ -0,0 +1,132 @@
+  private static final String UNIX_LAUNCHER_SCRIPT = "sonarlint-backend";
+  private static SonarLintRpcClient client;
+  private static Process process;
+
+  private SloopLauncher() {
+    // don't instantiate
+  }
+
+  @Nullable
+  public static SonarLintRpcServer startSonarLintRpcServer(String distArchivePath, SonarLintRpcClient clientArg) {
+    return startAndGetSonarLintRpcServer(distArchivePath, clientArg, "");
+  }
+
+  @Nullable
+  public static SonarLintRpcServer startSonarLintRpcServerWithJre(String distArchivePath, SonarLintRpcClient clientArg, String jrePath) {
+    return startAndGetSonarLintRpcServer(distArchivePath, clientArg, jrePath);
+  }
+
+  private static SonarLintRpcServer startAndGetSonarLintRpcServer(String distArchivePath, SonarLintRpcClient clientArg, String jrePath) {
+    try {
+      SloopLauncher.client = clientArg;
+      String outputDirPath = unarchiveDistribution(distArchivePath);
+      return execute(outputDirPath, jrePath);
+    } catch (Exception e) {
+      client.log(new LogParams(LogLevel.ERROR, e.getMessage(), null));
+    }
+    return null;
+  }
+
+
+  private static SonarLintRpcServer execute(String zipOutputDirPath, String jrePath) {
+    var binDirPath = zipOutputDirPath + File.separator + "bin";
+    List<String> commands = new ArrayList<>();
+    try {
+      if (SystemUtils.IS_OS_WINDOWS) {
+        commands.add("cmd.exe");
+        commands.add("/c");
+        commands.add(WIN_LAUNCHER_SCRIPT);
+      } else {
+        commands.add("sh");
+        commands.add(UNIX_LAUNCHER_SCRIPT);
+      }
+      if (!jrePath.isEmpty()) {
+        commands.add("-j");
+        commands.add(jrePath);
+      }
+
+      var processBuilder = new ProcessBuilder(commands);
+      processBuilder.directory(new File(binDirPath));
+      processBuilder.redirectOutput(ProcessBuilder.Redirect.PIPE);
+      processBuilder.redirectInput(ProcessBuilder.Redirect.PIPE);
+      process = processBuilder.start();
+
+      // use process.getInputStream() as an input for the client
+      var serverToClientInputStream = process.getInputStream();
+      // use process.getOutputStream() as the standard input of a subprocess that can be written to
+      var clientToServerOutputStream = process.getOutputStream();
+      var clientLauncher = new ClientJsonRpcLauncher(serverToClientInputStream, clientToServerOutputStream, client);
+
+      return clientLauncher.getServerProxy();
+    } catch (Exception e) {
+      client.log(new LogParams(LogLevel.WARN, e.getMessage(), ""));
+    }
+    return null;
+  }
+
+  public static int waitFor() throws InterruptedException {
+    if (process.waitFor(1, TimeUnit.MINUTES)){
+      return process.exitValue();
+    } else {
+      process.destroyForcibly();
+      return -1;
+    }
+  }
+
+  private static String unarchiveDistribution(String distArchivePath) {
+    var outputDirPath = Paths.get("").resolve("out").toAbsolutePath().toString();
```

Comment:
Not great to not control where the backend is extracted. In fact I don't think it should be the responsibility of this class to deal with the extraction. I guess IDEs will try to only extract once, to avoid the cost. Similar to what we do with Omnisharp, this class should probably take as argument the path where the backend is extracted.

Reply:
Addressed in [3a40e35](https://github.com/SonarSource/sonarlint-core/pull/765/commits/3a40e35e93ba7a3f6bf6823de2bbb829b72b9fc4)